src/ORM/AssociationCollection.php

```diff
@@ -78,6 +80,10 @@ public function add(string $alias, Association $association): Association
     {
         [, $alias] = pluginSplit($alias);
 
+        if (isset($this->_items[$alias])) {
+            throw new CakeException(sprintf('Association alias `%s` is already set.', $alias));
```

Comment:
Generally I'm in favour of this, but we should be prepared for this to impact userland plugins a bit.

Reply:
Do plugins stomp each other? Do we need to allow plugins to set if they're stompable? That would depend on order of load so maybe it's a useless concept.

Reply:
They *could* and that likely means it is happening somewhere in the wild. If this becomes a blocker we can always revisit it.

Reply:
I am willing to risk a few support tickets, if things get too bad we can always revert :)

Reply:
Based on the queries I have seen on our support channels, hidden duplication causes a lot of grief to users.

Reply:
ok, I'm onboard with the change :+1: